app/models.py

```diff
@@ -0,0 +1,176 @@
+
+    def __init__(self):
+        '''Create an empty database.'''
+
+        self.users = []
+        self.products = []
+        self.sales = []
+
+    def drop(self):
+        '''Drop entire database.'''
+
+        self.__init__()
+
+
+db = DB()
+
+
+
+class Base():
+    '''Base class to be inherited by other models.'''
+
+    def save(self):
+        '''Add object to database.'''
+        try:
+            self.id = getattr(db, self.tablename)[-1]['id']+1
+        except IndexError:
+            self.id = 1
+        current = self.current()
+        getattr(db, self.tablename).append(current)
+        return self.view()
+
+
+    def view(self):
+        '''View object as a dictionary.'''
+
+        return self.current
+
+
+class Product(Base):
+    '''Product model.'''
+
+    def __init__(self, name, price):
+        '''Initialize a product.'''
+
+        self.id = 0
+        self.name = name
+        self.price = price
+        self.tablename = 'products'
+
+    def current(self):
+        current = {
+            "name": self.name,
+            "price": self.price,
+            "id": self.id
+        }
+        return current
+
+    def view(self):
+        '''View a user's information.'''
+
+        return {
+            "name": self.name,
+            "price": self.price,
+            "id": self.id
+        }
+
+class Sale(Base):
+    '''Sales model.'''
+
+    def __init__(self, name, price):
+        '''Initialize a sale.'''
+
+        self.id = 0
+        self.name = name
+        self.price = price
+        self.tablename = 'sales'
+
+    def current(self):
+        current = {
+            "name": self.name,
+            "price": self.price,
+            "id": self.id
+        }
+        return current
+
+    def view(self):
+        '''View a user's information.'''
+
+        return {
+            "name": self.name,
+            "price": self.price,
+            "id": self.id
+        }
+
+
+
+class User(Base):
+    '''User model.'''
+
+    def __init__(self, username, password, email, isAdmin, isAttendant, id=None):
+        '''Initialize a user.'''
+
+        self.id = 0
+        self.username = username
+        self.email = email
+        self.password = self.make_hash(password)
+        if str(isAdmin) == 'True':
+            self.isAdmin = True
+        else:
+            self.isAdmin = False
+        if str(isAttendant) == 'True':
+            self.isAttendant = True
+        else:
+            self.isAttendant = False
+        self.tablename = 'users'
+
+
+    def current(self):
+        current = {
+            'username': self.username,
+            'email': self.email,
+            'password': self.password,
+            'isAttendant': self.isAttendant,
+            'isAdmin': self.isAdmin,
+            'id': self.id
+        }
+        return current
+
+    def view(self):
+        '''View a user's information.'''
+
+        return {
+            'username': self.username,
+            'email': self.email,
```

Comment:
Why are you using camel casing in your naming?

Reply:
Dan i really do not know much about camel case vs snake case naming.So i did what sufficed,but kindly point me in the right direction.

Reply:
First, if you do not know ask before implementing.

Reply:
Read [this](https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions)

Reply:
Thanks for the resource.I have changed to snake casing naming.